File: convert_image.py
import os
from PIL import Image

src_path = r"C:\Users\Admin\.gemini\antigravity\brain\c4941793-2885-42d4-bb63-1c2773802942\media__1782812062914.png"
dest_path = r"c:\Users\Admin\Desktop\virtual reality\person_jane.webp"

with Image.open(src_path) as img:
    q = 90
    img.save(dest_path, "webp", quality=q)
    while os.path.getsize(dest_path) > 100 * 1024 and q > 10:
        q -= 10
        img.save(dest_path, "webp", quality=q)

print(f"Saved to {dest_path} with size {os.path.getsize(dest_path) / 1024:.2f} KB")

File: convert_image.js
const sharp = require('sharp');
const fs = require('fs');

const srcPath = "C:\\Users\\Admin\\.gemini\\antigravity\\brain\\c4941793-2885-42d4-bb63-1c2773802942\\jane_doe_portrait_1782814477693.png";
const destPath = "c:\\Users\\Admin\\Desktop\\virtual reality\\person_jane.webp";

async function convertImage() {
  let quality = 90;
  let buffer;
  let done = false;

  while (!done && quality > 10) {
    buffer = await sharp(srcPath).webp({ quality: quality }).toBuffer();
    if (buffer.length <= 100 * 1024) {
      done = true;
    } else {
      quality -= 10;
    }
  }
  
  if (buffer) {
    fs.writeFileSync(destPath, buffer);
    console.log(`Saved to ${destPath} with quality ${quality}, size: ${(buffer.length / 1024).toFixed(2)} KB`);
  }
}

convertImage().catch(console.error);
